src/sys/socket/mod.rs

```diff
@@ -261,6 +261,21 @@ impl Ipv6MembershipRequest {
     }
 }
 
+/// SockExtendedErr is the struct returned when IP_RECVERR is used to
+/// receive asyncronous IP errors.
+#[cfg(any(target_os = "android", target_os = "linux"))]
+#[repr(C)]
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub struct SockExtendedErr {
```

Comment:
Is this struct defined anywhere in the C headers?  If so, you should add its definition to the libc crate, and reference that here.  Nix shouldn't be in the business of repeating the headers.

Reply:
It's defined in https://elixir.bootlin.com/linux/v5.4.1/source/include/uapi/linux/errqueue.h

I read the libc crate's RFC https://github.com/rust-lang/rfcs/blob/master/text/1291-promote-libc.md and it wasn't clear to me if this struct belonged there or not.

I am happy to open a PR there first and define it there.

Reply:
Yes, please submit a PR to libc.